app/controllers/spots_controller.rb

```diff
@@ -10,14 +10,29 @@ def new
   def create
     @spot = Spot.new(spot_params)
     if @spot.save
-      flash[:notice] = "Spot added."
+      flash[:success] = "Spot added."
       redirect_to spots_path
     else
-      flash[:notice] = "Spot was not saved."
+      flash[:alert] = @spot.errors.full_messages.join(".  ")
       render :new
     end
   end
 
+  def show
+    @spot = Spot.find(params[:id])
+    @reviews = @spot.reviews
+  end
+
+  def edit
+    @spot = Spot.find(params[:id])
+  end
+
+  def update
+    @spot = Spot.find(params[:id])
+    @spot.update!(spot_params)
```

Comment:
This should have a bang, we never want noisy failure on the user side. Make an `if` statement like in `create` and display error messages if the save isn't successful